providers/smsmode/src/lib/smsmode.provider.ts

```diff
@@ -0,0 +1,61 @@
+import {
+  ChannelTypeEnum,
+  ISendMessageSuccessResponse,
+  ISmsOptions,
+  ISmsProvider,
+} from '@novu/stateless';
+import { ProxyAgent } from 'proxy-agent';
+import 'cross-fetch';
+
+declare global {
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  interface RequestInit {
+    agent: ProxyAgent;
+  }
+}
```

Comment:
@scopsy shall we allow `proxy-agent` and `cross-fetch` in place of `axios`?

Reply:
I used this two for avoiding 502 proxy issue that axios have. You can see my other PR about proxy on sendinblue. Also, I've used it in brevo sms provider.